### ISSUE TITLE
NetInfo.isConnected always returns true after connection events

### ISSUE THREAD

**Issue by achipa** [microsoft/react-native-windows#1941]
### Environment

1. `react-native -v`:
  react-native-cli: 2.0.1
  react-native: 0.55.4
2. `npm ls rnpm-plugin-windows`: 
  rnpm-plugin-windows@0.2.8
3. `npm ls react-native-windows`:
  react-native-windows@0.55.0-rc.0
4. `node -v`:
  v10.8.0
5. `npm -v`:
  4.6.1
6. `yarn --version`
  1.9.4

Then, specify:

- Target Platform: UWP

- Target Platform Version(s): 10.0.14393

- Target Device(s): Desktop

- Development Operating System: Windows

- Visual Studio Version: VS2017

- Build Configuration: Debug

### Steps to Reproduce

Create React Native Windows UWP application
add NetInfo listener as per the NetInfo Documentation at https://facebook.github.io/react-native/docs/netinfo
start application (while online through a WiFi connection)
go offline
go online
turn flight mode on
turn flight mode off

### Expected Behavior

NetInfo to reflect actual connectivity status:

[JS Info] First, is online
[JS Info] Then, is offline
[JS Info] Then, is online
[JS Info] Then, is offline
[JS Info] Then, is online

### Actual Behavior

For the scenario above, the code returns

[JS Info] First, is online
[JS Info] Then, is online
[JS Info] Then, is online
[JS Info] Then, is online
[JS Info] Then, is online

The connectionChange event seem to get triggered correctly, but the state of isConnected does not get returned/set correctly.

### Reproducible Demo

https://snack.expo.io/BJkjAymIQ

```
NetInfo.isConnected.fetch().then(isConnected => {
  console.log('First, is ' + (isConnected ? 'online' : 'offline'));
});
function handleFirstConnectivityChange(isConnected) {
  console.log('Then, is ' + (isConnected ? 'online' : 'offline'));
}
NetInfo.isConnected.addEventListener(
  'connectionChange',
  handleFirstConnectivityChange
);
```

**Comment by rozele** [comment]
Thanks for filing! @piyush965 modified NetInfo to the non-deprecated API - can you check if this resolves the issue for you? If so I can prioritize the review and get that merged.

**Comment by achipa** [comment]
No, unfortunately that does not resolve the core issue.

There is _some_ difference in terms of NetInfo.getConnectionInfo behaving slightly differently. In the master branch, it's returning an empty object, while on that branch it returns

`
[JS Info] {"type":"LocalAccess","effectiveType":"unknown"}
`

If the device is offline (both flight mode and simply not connected), and 

`
[JS Info] {"type":"InternetAccess","effectiveType":"unknown"}
`

if the device is online. This, while a potential workaround, seems wrong as it's not in line with what the React Native documentation says what "type" should be:

ConnectionType:
none - device is offline
wifi - device is online and connected via wifi, or is the iOS simulator
cellular - device is connected via Edge, 3G, WiMax, or LTE
unknown - error case and the network status is unknown

**Comment by achipa** [comment]
It seems the issue at least in part stems from the interpretation of the connection profile. The values are, according to https://docs.microsoft.com/en-us/uwp/api/windows.networking.connectivity.networkconnectivitylevel

InternetAccess | 3 | Local and Internet access.
ConstrainedInternetAccess | 2 | Limited internet access. This value indicates captive portal connectivity, LocalAccess | 1 | Local network access only.
None | 0 | No connectivity.

The confusion comes from everything non-zero being interpreted as isConnected = true, which is technically true (there might be local interfaces active, and Windows doesn't disable those even in flight mode), so this definition of "offline" is likely not what most developers would have in mind.

**Comment by achipa** [comment]
The culprit turned out to be the virtual Ethernet switch in my case:

![networking](https://user-images.githubusercontent.com/822934/44337810-d239d580-a473-11e8-99e6-adbc02578731.png)

The point being that unlike "regular" interfaces, Windows doesn't disable/shut down these when going to flight mode (again, I presume because technically this is a form of local networking), and then the conclusion is that you're still connected (even if it is just Hyper-V or sorts).

@piyush965 I'm getting type: "LocalAccess" when offline. It's not wrong (sort of), but it is misleading (in combination with isConnected still being true).

@rozele I'm happy to supply a patch that would interpret LocalAccess as isConnected: false (NetInfo should be about data being able the leave the device, not getting philosophical about local network interfaces), but @piyush965 's patch is a prerequisite for that (as that's how we get the LocalAccess info).

**Comment by piyush965** [comment]
We had the similar issue and we ended up using:
function handleFirstConnectivityChange(isConnected) {
  console.log('is Online ' + (isConnected ? OS.windows ? isConnected.type === 'InternetAccess' : isConnected.type !== 'none' && isConnected.type !== 'unknown'));
}

**Comment by achipa** [comment]
Yes, I'm using something of the kind as well, but that's IMHO a less-than-elegant workaround, the original point(s) still stand:

a) isConnected can be misleading for (most?) developers
b) properties w primitive types like boolean shouldn't have per-platform meaning (enums are a different story)

**Comment by GezimVeliu** [comment]
Did anyone come up with a solution?

We have the same problem, isConnected returns a true value and isConnected.type is undefined, isConnected.effectiveType is undefined.

**Comment by rozele** [comment]
Is there a less hacky workaround, or a way to bring fix the issues with #1864?

**Comment by mMarcos208** [comment]
I have the same problem

```
  NetInfo.fetch().then((state) => {
      setIsConnected(state.isConnected)
      console.log(state.isConnected)
    })
```

This code always return true in Android Emulator.

Version 6.0.0
